tests/testMoveFromSan.ts

```diff
@@ -0,0 +1,7 @@
+import { Chess } from '../cjsmin/src/chess';
+
+const chess = new Chess();
+const move = 'O-O-O';
+const result = chess.testMoveFromSan(move);
+
+console.log(`Result of moveFromSan for move ${move}:`, result);
```

Comment:
hmm, this is pretty concerning. Good job setting up the test case, probably should have pushed to have the full range of moves from the get go. Will take a look and see if I can debug

Reply:
looks like castling generally is broken. Good news is we can hack a fix if I can't easily debug where I broke it

Reply:
There's also a different bug bc I allow pieces in the board array to be null, before they could only be undefined